src/server/api/modules/recipes/service/schemas.ts

```diff
@@ -1,10 +1,15 @@
 import z from "zod"
 
 export const RecipeStepSchema = z
-  .union([z.string(), z.object({ text: z.string() })])
+  .union([
+    z.string(),
+    z.object({ "@type": z.string(), text: z.string() }),
+    z.object({ "@type": z.string(), text: z.string() }).array().nonempty()
```

Comment:
```suggestion
    z.object({ "@type": z.string(), text: z.string() }).array()
```

I would shy away from insisting that the array has to be non-empty - the user can always correct/add steps in Grocy once the recipe is imported, but this would cause the import to RB to fail entirely and rob them of this possibility.